File: articles/decision-science-vs-decision-theory/python/run_all_decision_theory_science_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Decision Science vs. Decision Theory."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "expected_utility_comparison.py",
    "bayesian_update_decision_rule.py",
    "minimax_regret_diagnostics.py",
    "satisficing_agent_model.py",
    "robust_adaptive_strategy_scan.py",
    "institutional_stress_testing.py",
    "decision_record_exporter.py",
    "decision_science_vs_decision_theory_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python decision theory/science workflows completed.")


if __name__ == "__main__":
    main()

File: articles/decision-science-vs-decision-theory/r/run_all_decision_theory_science_workflows.R
# run_all_decision_theory_science_workflows.R
# Run all R workflows for Decision Science vs. Decision Theory.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_science_vs_decision_theory_workflow.R",
  "normative_criteria_comparison.R",
  "regret_profile_diagnostics.R",
  "robustness_sensitivity_analysis.R",
  "probability_sensitivity_profiles.R",
  "decision_criteria_visualization.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R decision theory/science workflows completed.")
